sklearn/preprocessing/_encoders.py

```diff
@@ -0,0 +1,850 @@
+    array([[ 1.,  0.,  0.,  1.,  0.,  0.,  1.,  0.,  0.]])
+
+    See also
+    --------
+    sklearn.preprocessing.OrdinalEncoder : performs an ordinal (integer)
+      encoding of the categorical features.
+    sklearn.feature_extraction.DictVectorizer : performs a one-hot encoding of
+      dictionary items (also handles string-valued features).
+    sklearn.feature_extraction.FeatureHasher : performs an approximate one-hot
+      encoding of dictionary items or strings.
+    sklearn.preprocessing.LabelBinarizer : binarizes labels in a one-vs-all
+      fashion.
+    sklearn.preprocessing.MultiLabelBinarizer : transforms between iterable of
+      iterables and a multilabel format, e.g. a (samples x classes) binary
+      matrix indicating the presence of a class label.
+    sklearn.preprocessing.LabelEncoder : encodes labels with values between 0
+      and n_classes-1.
+    """
+    def __init__(self, n_values=None, categorical_features=None,
+                 categories='auto', sparse=True, dtype=np.float64,
+                 handle_unknown='error'):
+        self.categories = categories
+        self.sparse = sparse
+        self.dtype = dtype
+        self.handle_unknown = handle_unknown
+
+        if n_values is not None:
+            warnings.warn("Deprecated", DeprecationWarning)
+        else:
+            n_values = "auto"
+        self._deprecated_n_values = n_values
+
+        if categorical_features is not None:
+            warnings.warn("Deprecated", DeprecationWarning)
+        else:
+            categorical_features = "all"
+        self._deprecated_categorical_features = categorical_features
+
+    # Deprecated keywords
+
+    @property
+    def n_values(self):
+        warnings.warn("Deprecated", DeprecationWarning)
+        return self._deprecated_n_values
+
+    @n_values.setter
+    def n_values(self, value):
+        warnings.warn("Deprecated", DeprecationWarning)
+        self._deprecated_n_values = value
+
+    @property
+    def categorical_features(self):
+        warnings.warn("Deprecated", DeprecationWarning)
+        return self._deprecated_categorical_features
+
+    @categorical_features.setter
+    def categorical_features(self, value):
+        warnings.warn("Deprecated", DeprecationWarning)
+        self._deprecated_categorical_features = value
+
+    # Deprecated attributes
+
+    @property
+    def active_features_(self):
+        check_is_fitted(self, 'categories_')
+        warnings.warn("Deprecated", DeprecationWarning)
+        return self._active_features_
+
+    @property
+    def feature_indices_(self):
+        check_is_fitted(self, 'categories_')
+        warnings.warn("Deprecated", DeprecationWarning)
+        return self._feature_indices_
+
+    @property
+    def n_values_(self):
+        check_is_fitted(self, 'categories_')
+        warnings.warn("Deprecated", DeprecationWarning)
+        return self._n_values_
+
+    def _handle_deprecations(self, X):
+
+        if self.categories != 'auto':
```

Comment:
Let's make `self.categories='legacy'` by default, so that the user can select `auto` explicitly.

Reply:
Yesterday I made a commit to make the default `categories=None` for the same purpose, but only pushed it now (the last commit). 
I personally find that cleaner, as `'legacy'` would not give the correct meaning if you are using string data.